package/sbp_rtcm3_bridge/src/sbp_rtcm3.c

```diff
@@ -509,3 +511,32 @@ void gps_time_callback(u16 sender_id, u8 len, u8 msg[], void *context)
   time_from_rover_obs.tow = time->tow;
   time_from_rover_obs.ns_residual = time->ns_residual;
 }
+
+void sbp_obs_callback(u16 sender_id, u8 len, u8 msg[], void *context)
+{
+  (void) context;
+
+  /* An SBP sender ID of zero means that the messages are relayed observations
+   * from the console. We don't want to use them */
+  if (sender_id == 0) {
+    return;
+  }
+
+  observation_header_t* obs_header = (observation_header_t*)msg;
+  gps_time_nano_t tor = obs_header->t;
+
+  /* Calculate the number of observations in this message by looking at the SBP
+   * `len` field. */
+  u8 obs_in_msg = (len - sizeof(observation_header_t)) / sizeof(packed_obs_content_t);
+
+  /* Pull out the contents of the message. */
+  packed_obs_content_t *obs = (packed_obs_content_t *)(msg + sizeof(observation_header_t));
+  for (u8 i=0; i<obs_in_msg; i++) {
+    gnss_signal16_t sid = obs[i].sid;
+
+    /* We're only interested in the pseudorange observation */
+    double pseudorange = ((double)obs[i].P) / MSG_OBS_P_MULTIPLIER;
+
+    update_pseudorange_ambiguity(sid, tor, pseudorange);
```

Comment:
I think the RTCM amb is based on the L1 code, so we should probably only update in this case (though it won't matter if they are within a ms of each other).